share/spack/templates/modules/modulefile.lua

```diff
@@ -82,6 +82,8 @@ remove_path("{{ cmd.name }}", "{{ cmd.value }}", "{{ cmd.separator }}")
 setenv("{{ cmd.name }}", "{{ cmd.value }}")
 {% elif command_name == 'UnsetEnv' %}
 unsetenv("{{ cmd.name }}")
+{% elif command_name == 'PushEnv' %}
```

Comment:
Would it work for you, instead of making `pushenv` an `EnvironmentModifications` command, if we had a configuration for `lmod` similar to:
```yaml
modules:
  lmod:
    commands:
      set: pushenv   
```
that permits to select how to treat an `EnvironmentModification.set()` command in the context of `lmod` module file generation?